pyCalculations/fieldtracer.py

```diff
@@ -190,89 +190,140 @@ def static_field_tracer_3d( vlsvReader, coord_list, max_iterations, dx, directio
        param dx:              One iteration step length [meters] (ex. dx=1e4 for typical applications)
        keyword direction:     '+' or '-' or '+-' Follow field in the plus direction, minus direction, or both
-       keyword fg:            The field grid variable to be traced (either a string or numpy array)
+       keyword grid_var:      Variable to be traced (A string)
                               options include:
-                                  fg = some string
+                                  grid_var = some string
                                       ex. fg='fg_b': B-field, fg='fg_e': E-field
                                       static_field_tracer_3d() will load the appropriate variable via the vlsvReader object
                                       NOTE: volumetric variables, with '_vol' suffix, may not work as intended. Use face-centered values: 'fg_b', 'fg_e' etc.
-                                  fg = some field-grid ("fg") array.          dimensions [dimx,dimy,dimz,3]
+                                  grid_var = some field-grid ("fg") array.          dimensions [dimx,dimy,dimz,3]
                                       ex. fg = vlsvobj.read_variable('fg_b')
                                       field grid data is already loaded externally using read_variable() method (see vlsvreader.py).
                                       If fg keyword is set this way, the input vlsvReader is only referred to for metadata (esp. grid dimensions)
-                                   
-      :returns:                  points_traced --- Traced coordinates (a list of lists of 3-element coordinate arrays)
+                                  grid_variable = 'vg_b_vol'
+
+      :returns:               fg:   points_traced --- Traced coordinates (a list of lists of 3-element coordinate arrays)
                                  ex. points_traced[2][5][1]: at 3rd tracing step [2], the 6th point [5], y-coordinate [1]
                                     note: Can convert output to a 3D numpy array if desired, with np.array(points_traced)
+                              vg:   points_traced --- a 3d numpy array [len(seed_coords)]
       EXAMPLE:            vlsvobj = pytools.vlsvfile.VlsvReader(vlsvfile) 
                           fg_b = vlsvobj.read_variable('fg_b')
                           traces = static_field_tracer_3d( vlsvobj, [[5e7,0,0], [0,0,5e7]], 10, 1e5, direction='+', fg = fg_b )
    '''
 
-   # standardize input (a list of 3-element arrays/lists)
-   if type(coord_list[0]) not in [list, np.ndarray]:
-      coord_list = [coord_list]
+   # Standardize input: (N,3) np.array
+   if type(seed_coords) != np.ndarray:
+      raise TypeError("Please give a numpy array.")
+
+   # Cache and read variables:
+   vg = None
+   fg = None
+
+   if type(grid_var) == str:
+      if grid_var.startswith("fg"):
+         fg = grid_var
+
+      elif grid_var.startswith("vg"):   
```

Comment:
Hm. We may actually want to trace variables with names like "proton/vg_v". string,split('/') and check if any of the substrings starts with vg?